src/AutoRest.CSharp/Common/Generation/Writers/RequestWriterHelpers.cs

```diff
@@ -190,13 +197,23 @@ public static void WriteRequestCreation(CodeWriter writer, RestClientMethod clie
                 }
 
                 if (writeUserAgentOverride)
-                    writer.Line($"message.SetProperty(\"UserAgentOverride\", _userAgent);");
+                {
+                    writer.Line($"{message}.SetProperty(\"UserAgentOverride\", _userAgent);");
+                }
+
+                if (responseClassifierType != default)
+                {
+                    writer.Line($"{message}.{nameof(HttpMessage.ResponseClassifier)} = {responseClassifierType}.Instance;");
+                }
 
                 writer.Line($"return {message};");
             }
             writer.Line();
         }
 
+        private static ReferenceOrConstant FixFieldReference(IReadOnlyDictionary<string, string>? fields, ReferenceOrConstant value) =>
```

Comment:
What case does this handle?

Reply:
If parameter is actually has been saved as a field, it swaps the name of parameter with the name of the field. E.g. `endpoint` -> `_endpoint`

Reply:
Let me try

Reply:
https://github.com/Azure/autorest.csharp/issues/1552